amity/amity.py

```diff
@@ -0,0 +1,255 @@
+                                floor=floor,
+                                no_of_occupants = self.no_of_occupants,
+                                capacity=self.office_capacity
+                             )
+        else:
+            new_room = LivingSpace(
+                                    identifier=identifier, 
+                                    name=name, 
+                                    floor=floor,
+                                    no_of_occupants = self.no_of_occupants,
+                                    capacity=self.living_space_capacity
+                                  )
+        return new_room
+
+    def create_person(self, identifier, first_name, last_name, employeeType, 
+            wants_accommodation, rooms):
+        """Create Person object"""
+        if employeeType == "staff":
+            new_person = Staff(identifier=identifier, first_name=first_name, 
+                last_name=last_name)
+        else:
+            new_person = Fellow(identifier=identifier, first_name=first_name, 
+                last_name=last_name)
+        return self.randomly_allocate_rooms(
+                                              new_person, 
+                                              wants_accommodation,
+                                              rooms
+                                           )
+
+    def randomly_allocate_rooms(self, person, wants_accommodation, rooms):
+        """Allocate person to room
+        Both staff and fellow are assigned office spaces
+        Only fellow who indicates interest is assigned living space
+        Args:
+        person -- the person of interest
+        wants_accommodation -- indicates if person wants accommodation or not
+        rooms -- list of all created rooms
+        Returns:
+        a call to add_persons_to_room method 
+        """
+        allocated_space  = []
+        if self.check_room_type_has_vacancy(rooms, "office"):
+            allocated_space.append(self.randomly_pick_a_room(rooms, "office"))
+        if person.type == "fellow" and wants_accommodation == "Y" \
+            and self.check_room_type_has_vacancy(rooms, "living space"):
+            allocated_space.append(self.randomly_pick_a_room(rooms, 
+                                                             "living space"
+                                                            ))
+        return self.add_person_to_rooms(person, allocated_space)
+        
+    def check_room_type_has_vacancy(self, rooms, room_type):
+        """
+        Args:
+        rooms -- all rooms in database
+        room_type -- the room type of interest
+        Returns:
+        True -- if there is a vacant room of a particular type. 
+        False -- if there is none
+        """
+        for room in rooms:
+            if self.check_that_room_is_vacant(room, room_type):
+                return True
+        print "There is no vacant {0} currently in database!!!". \
```

Comment:
**Code Quality:** For the `Amity` class, either raise an exception or return something. Leave the other parts for the CLI interface.